cache/s3.go

```diff
@@ -109,34 +106,48 @@ func (c S3Cache) GetArtifact(artifactID string) (types.ArtifactInfo, error) {
 	if err != nil {
 		return types.ArtifactInfo{}, xerrors.Errorf("JSON unmarshal error: %w", err)
 	}
-
 	return info, nil
 }
 
+func (c S3Cache) getIndex(key string, keyType string) error {
+	_, err := c.s3Client.HeadObject(&s3.HeadObjectInput{Key: aws.String(fmt.Sprintf("%s/%s/%s.index", keyType, c.prefix, key))})
+	if err != nil {
+		return xerrors.Errorf("failed to get index from the cache: %w", err)
+	}
+	return nil
+}
+
 func (c S3Cache) MissingBlobs(artifactID string, blobIDs []string) (bool, []string, error) {
 	var missingArtifact bool
 	var missingBlobIDs []string
 	for _, blobID := range blobIDs {
-		blobInfo, err := c.GetBlob(blobID)
+		err := c.getIndex(blobID, blobBucket)
 		if err != nil {
 			// error means cache missed blob info
 			missingBlobIDs = append(missingBlobIDs, blobID)
 			continue
 		}
+		blobInfo, err := c.GetBlob(blobID)
+		if err != nil {
+			return true, missingBlobIDs, xerrors.Errorf("failed to get blob from the cache: %w", err)
```

Comment:
You want to return an error instead of `continue` because this should never happen, right?

```suggestion
			return true, missingBlobIDs, xerrors.Errorf("the blob object (%s) doesn't exist in S3 even though the index file exists: %w", blobID, err)
```